network/p2p/network_test.go

```diff
@@ -25,289 +20,293 @@ import (
+	require := require.New(t)
+	ctx := context.Background()
 
-				require.NoError(t, client.AppRequest(context.Background(), set.Of(nodeID), request, callback))
-			},
-		},
-		{
-			name: "cross-chain app request",
-			requestFunc: func(t *testing.T, network *Network, client *Client, sender *common.SenderTest, handler *mocks.MockHandler, wg *sync.WaitGroup) {
-				chainID := ids.GenerateTestID()
-				sender.SendCrossChainAppRequestF = func(ctx context.Context, chainID ids.ID, requestID uint32, request []byte) {
-					go func() {
-						require.NoError(t, network.CrossChainAppRequest(ctx, chainID, requestID, time.Time{}, request))
-					}()
-				}
-				sender.SendCrossChainAppResponseF = func(ctx context.Context, chainID ids.ID, requestID uint32, response []byte) {
-					go func() {
-						ctx = context.WithValue(ctx, ctxKey, ctxVal)
-						require.NoError(t, network.CrossChainAppResponse(ctx, chainID, requestID, response))
-					}()
-				}
-				handler.EXPECT().
-					CrossChainAppRequest(context.Background(), chainID, gomock.Any(), request).
-					DoAndReturn(func(context.Context, ids.ID, time.Time, []byte) ([]byte, error) {
-						return response, nil
-					})
-
-				callback := func(ctx context.Context, actualChainID ids.ID, actualResponse []byte, err error) {
-					defer wg.Done()
-					require.NoError(t, err)
-					require.Equal(t, ctxVal, ctx.Value(ctxKey))
-					require.Equal(t, chainID, actualChainID)
-					require.Equal(t, response, actualResponse)
-				}
+	sender := common.FakeSender{
+		SentAppRequest: make(chan []byte, 1),
+	}
+	network := NewNetwork(logging.NoLog{}, sender, prometheus.NewRegistry(), "")
 
-				require.NoError(t, client.CrossChainAppRequest(context.Background(), chainID, request, callback))
-			},
-		},
-		{
-			name: "cross-chain app request failed",
-			requestFunc: func(t *testing.T, network *Network, client *Client, sender *common.SenderTest, handler *mocks.MockHandler, wg *sync.WaitGroup) {
-				sender.SendCrossChainAppRequestF = func(ctx context.Context, chainID ids.ID, requestID uint32, request []byte) {
-					go func() {
-						require.NoError(t, network.CrossChainAppRequestFailed(ctx, chainID, requestID))
-					}()
-				}
+	client, err := network.NewAppProtocol(handlerID, &NoOpHandler{})
+	require.NoError(err)
 
-				callback := func(_ context.Context, actualChainID ids.ID, actualResponse []byte, err error) {
-					defer wg.Done()
+	wantNodeID := ids.GenerateTestNodeID()
+	done := make(chan struct{})
 
-					require.ErrorIs(t, err, ErrAppRequestFailed)
-					require.Equal(t, chainID, actualChainID)
-					require.Nil(t, actualResponse)
-				}
+	callback := func(_ context.Context, gotNodeID ids.NodeID, gotResponse []byte, err error) {
+		require.Equal(wantNodeID, gotNodeID)
+		require.ErrorIs(err, ErrAppRequestFailed)
+		require.Nil(gotResponse)
 
-				require.NoError(t, client.CrossChainAppRequest(context.Background(), chainID, request, callback))
-			},
-		},
-		{
-			name: "app gossip",
```

Comment:
Looks like we're not testing AppGossip/AppGossipSpecific anymore?

Reply:
Ugh you're right.... added `TestMessageRouting`.